src/l500/l500-color.cpp

```diff
@@ -240,7 +240,7 @@ namespace librealsense
             auto model = intrinsic.resolution.intrinsic_resolution[i];
             if( model.height == height && model.width == width )
             {
-                rs2_intrinsics intrinsics;
+                rs2_intrinsics intrinsics = {};
```

Comment:
I understand this change is to remove the uninitialized value used.
does this result to all members equal to zero?
I am not sure this will be the same behavior with any compiler

Reply:
It should always be the same (if compiler is according to standard.

Empty braces are [value initialization](https://en.cppreference.com/w/cpp/language/value_initialization) but since it is a struct [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization) will take place and
> if the element is not a reference, the element is copy-initialized from an empty initializer list

Reply:
What does it mean? there are no values in the initializer list, how this removes the CppCheck warning?

Reply:
The members are native types and will be zeroed.
Can also see [here](https://stackoverflow.com/questions/3803153/what-are-primitive-types-default-initialized-to-in-c)

Reply:
So @OhadMeir , you are saying that is I have a native only struct, I can create the instanse with = {} and it will surly be zeroed?

Reply:
Yes.
Tested with VS2019 and gcc